### ISSUE TITLE
@types/audioworklet + DOM API

### ISSUE THREAD

**Issue by h-a-n-n-e-s** [microsoft/TypeScript-DOM-lib-generator#1574]
Is there a proper way to use [@types/audioworklet](https://www.npmjs.com/package/@types/audioworklet) while keeping `"lib": ["DOM",...]` ?
(In the installation notes it says: Update your _tsconfig.json_ to avoid clashing with the DOM APIs...  You should remove "dom")

For doing front-end development while using `AudioWorkletProcessor` that would be really helpful.

**Comment by saschanaz** [comment]
What's the use case here? Are you creating a script that works as a normal script and also as a worklet script?

**Comment by h-a-n-n-e-s** [comment]
I'm writing a web app (that's what I need the DOM for) which uses webAudio including the [AudioWorkletProcessor](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletProcessor) class. The types of that class are unknown by default (I wonder why?) so I have to use a JS file to make use of the class. I would like like to use TS for that class as well.

**Comment by saschanaz** [comment]
That doesn't answer the question. Can you move your worklet script to a subdirectory and add another tsconfig.json to apply another set of "lib"? Why would you need both in one tsconfig?

**Comment by h-a-n-n-e-s** [comment]
So I guess it is only possible with "project references".

**Coming back to the original issue**, I just tried to add `@types-audioworklet` without deleting `"DOM"` from `"lib"` in the (one and only) `tsconfig.json` and everything works fine!

Could somebody give an example of how a "clashing with the DOM APIs" could happen?

**Comment by Krivega** [comment]
> So I guess it is only possible with "project references".
> 
> **Coming back to the original issue**, I just tried to add `@types-audioworklet` without deleting `"DOM"` from `"lib"` in the (one and only) `tsconfig.json` and everything works fine!
> 
> Could somebody give an example of how a "clashing with the DOM APIs" could happen?


```JSON 
"lib": ["DOM", "ESNext"],
"compilerOptions": {
    "types": [ "audioworklet"]
  },
  ``` 
  
![image](https://github.com/microsoft/TypeScript-DOM-lib-generator/assets/1925918/2b8de98d-6fcc-43ac-84f8-5a2d6c4238ae)

**Comment by h-a-n-n-e-s** [comment]
Sorry for asking very basic questions, but why do I need `"types": [ "audioworklet"]` in `"compilerOptions"`? For me everything works fine without it. (And with it nothing changes.)

**Comment by Bashamega** [comment]
Hello @h-a-n-n-e-s 
Sorry for the late reply

The `"types": ["audioworklet"]` entry in your `tsconfig.json` file specifies that TypeScript should include type definitions for the `audioworklet` feature, which is part of the Web Audio API.

If everything works fine without it, it's likely because:

1. **TypeScript's Default Behavior**: TypeScript already has basic type definitions for common JavaScript APIs, including the Web Audio API. So, it's possible that TypeScript automatically includes these definitions without needing to explicitly list them in the `types` array.

2. **No Direct Usage**: If you aren't using the `AudioWorklet` directly in your code (or aren't using any features that require it), you wouldn't notice any difference whether or not it's explicitly listed in the `types` array.

In short, you only need `"audioworklet"` in the `types` array if you are working directly with the Web Audio API's `AudioWorklet` and want to ensure TypeScript has type definitions for it. If everything works without it, it just means TypeScript is able to resolve types for it without extra configuration in your case.
